--- awsetbg.old	2009-12-16 23:33:12.000000000 +0100
+++ awsetbg.new	2009-12-16 23:31:05.000000000 +0100
@@ -53,7 +53,7 @@
 fi
 
 # The wallpapersetter is selected in this order
-wpsetters="${wpsetters:=Esetroot habak wmsetbg feh hsetroot chbg display qiv xv xsri xli xsetbg xsetroot}" # broken icewmbg'
+wpsetters="${wpsetters:=feh Esetroot habak wmsetbg hsetroot chbg display qiv xv xsri xli xsetbg xsetroot}" # broken icewmbg'
 lastwallpaper="${CONFIG_DIR}/lastwallpaper"
 
 
